RL_tictactoe/experiments.py

```diff
@@ -0,0 +1,494 @@
+
+
+def plot_self_random_comparison_scores():
+    # Build plots
+    PATH = 'output/self_dummy_comparison_scores.xlsx'
+    sheets = pd.ExcelFile(PATH).sheet_names
+    
+    f = plt.figure()
+    legend_list = [] 
+    for sheet in sheets:
+        df_temp = pd.read_excel(PATH, sheet)
+        plt.plot(df_temp.index, df_temp['losses'])
+
+        legend_list.append(sheet)
+
+    legend_list = ['Self-Play vs Meta', 'Self-Play vs Self-Play', 'Self-Play vs Random',
+                   'Random vs Meta', 'Random vs Self-Play', 'Random vs Random']
+    # plot formatting
+    plt.title('Self-Play vs Random Training: Learning Rate', fontsize=18)
+    plt.legend(legend_list, fontsize=13)
+    plt.xlabel('Iterations (000s)', fontsize=13)
+    plt.ylabel('Loss (%)', fontsize=13)
+
+    # show plot
+    plt.show()
+
+
+
+def exploration_vs_winrate_comparison_scores():
+    # Build Agent
+    agent = Agent(1, 0)
+
+    # Load Meta Agent
+    policy = pickle.load(open('policies/meta_agent.p', 'rb'))
+    meta_agent = Agent(2, 0)
+    meta_agent.set_policy(policy)
+
+    # train test
+    df_scores = pd.DataFrame()
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~`
+
+    # Build Agents
+    a1 = Agent(1, 0.0)
+    a2 = Agent(1, 0.2)
+    a3 = Agent(1, 0.4)
+    a4 = Agent(1, 0.6)
+    a5 = Agent(1, 0.8)
+    a6 = Agent(1, 1.0)
+
+    # Build agent_list
+    exploration_list = [0, 0.2, 0.4, 0.6, 0.8, 1]
+    agent_list = [a1, a2, a3, a4, a5, a6]
+    test_list = [meta_agent] * 6
+
+    # Training Testing Iterations
+    df1, df2, df3, df4, df5, df6 = pd.DataFrame(), pd.DataFrame(), pd.DataFrame(), pd.DataFrame(), pd.DataFrame(), pd.DataFrame()
+
+    df_list = [df1, df2, df3, df4, df5, df6]
+
+    # iterate
+    for i in range(100):
+        print('Iteration: ', i)
+        # train
+        for a, e in zip(agent_list, exploration_list):
+            a.set_exploration(e)
+        agent_list = train_parallel(1000, agent_list, test_list)[0]
+
+        # test
+        set_exploration(agent_list, 0)
+        scores = test_parallel(10000, agent_list, test_list)
+        
+        # scores
+        scores_dfs = [pd.DataFrame(s, index=[0]) for s in scores]
+        df_list = [df.append(s_df) for df, s_df in zip(df_list, scores_dfs)]
+
+        print(' ')
+    
+
+    # DataFrame Columns
+    for df, e in zip(df_list, exploration_list):
+        df.drop('ties', axis=1, inplace=True)
+        df.drop('losses', axis=1, inplace=True)
+        df.columns = [str(col) + '(' + str(e) + ')' for col in df.columns]
+        print(df)
+
+
+    # DataFrame Output
+    df_scores = pd.concat(df_list, axis=1)
+
+    df_scores.to_excel('output/Exploration Rate vs Learning Rate.xlsx', index=False)
+    
+# # MAIN ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+if __name__ == "__main__":
+
+    excel = pd.ExcelFile('output/selfplay_vs_random_training.xlsx')
+    sheets = excel.sheet_names
+
+
```

Comment:
Okay so there are some functions here that should be in experiments.py and some that should stay here. The ones that specifically run experiments that we need for our exercise should stay here. But all helper functions that generally run experiments should be moved to experiments.py. Also pls rename this horribly named file (my b) to run_experiments.py